os/board/common/partitions.c

```diff
@@ -380,7 +383,7 @@ int configure_mtd_partitions(struct mtd_dev_s *mtd, int minor, partition_info_t
 
 #ifdef CONFIG_BINARY_MANAGER
 		if (!strncmp(types, "kernel,", 7)) {
-			binary_manager_register_kpart(g_partno, partsize, partoffset * geo.blocksize);
+			binary_manager_register_kpart(g_partno, partsize, part_data.start_addr + partoffset * geo.blocksize);
```

Comment:
This is required only for binary manager?
I mean flash startaddress is stil 0x00000000 but binary manager should use 0x0a0000000 ?

Reply:
It seems that only binary manager is used now. If there is any hard-coded address for external flash access in realtek codes, they can user this value I think.

signing verification function has a parameter of actual flash address. So it should be set as actual address.

Reply:
Then I think both of prev code & current approach also wrong.
Binary Manager also should use pointer of mtd to read data from flash, without information of start address...
Anyway you can't change it right now, but make sure that current design is wrong.

Reply:
I agree with your opinion. BM already access flash with mtd driver. But signing API is served by realtek and it requires actual address so I made it like this.